### PR TITLE
+ added keyboard support

### PR DESCRIPTION
**Pull request by 1lyyfe**
+ added keyboard support for viewControllers which conform to shortFo…

+ update scrollview position when showing keyboard if needed

###  Summary

Describe the goal of this PR. Mention any related Issue numbers.

### Requirements (place an `x` in each `[ ]`)

* [ ] I've read and understood the [Contributing Guidelines](https://github.com/slackhq/PanModal/blob/master/CONTRIBUTING.md) and have done my best effort to follow them.
* [ ] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

* [ ] I've written tests to cover the new code and functionality included in this PR.

### CODE REVIEW

**Review by mashdup — APPROVED** [review]

**Comment by markdarling** [comment]
Should you not be de-observing the keyboard somewhere?

File: PanModal/Controller/PanModalPresentationController.swift [hunk 1]
```diff
@@ -234,12 +246,57 @@ open class PanModalPresentationController: UIPresentationController {
...
+    // MARK: - Actions
+    
+    /**
+     Update long form height and short form height when the keyboard is showing/scrolling and scroll to correct position above the keyboard if keyboard is showing
+     */
+    @objc func keyboardWillShow(notification: Notification) {
+        
+        //if the presented view is already at max height and is scrollable then scroll to above the keyboard when keyboard is shown if needed
+        
+        if let keyboardSize = (notification.userInfo?  [UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue {
```
Comment (markj) [inline PanModal/Controller/PanModalPresentationController.swift]: That's a bit crazy.. why not get it as a GSRect in the first place..
`let kbRect = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect,`

File: PanModal/Controller/PanModalPresentationController.swift [hunk 2]
```diff
@@ -234,12 +246,57 @@ open class PanModalPresentationController: UIPresentationController {
...
+            
+            if let scrollView = presentable?.panScrollable, pannedToMax {
+                scrollView.contentInset = UIEdgeInsets(top: 0.0, left: 0.0, bottom: keyboardSize.height, right: 0.0);
```
Comment (markj) [inline PanModal/Controller/PanModalPresentationController.swift]: This makes assumptions that there is no content inset for the scrollview in the first place.
You should be respecting what values are there already.

**Comment by markj** [comment]
I'm on the outside of this.. but associating this keyboard behaviour with something that MUST have a scrollview seems wrong.. what if you present a small VC that has two fields in but doesn't need a scrollview?

Wouldn't it be better to handle the keyboard position in dealing with where the child VC is presenting. Then you aren't modifying things you aren't in charge of (e.g. scrollview content insets)

File: PanModal/Controller/PanModalPresentationController.swift [hunk 3]
```diff
@@ -234,12 +246,57 @@ open class PanModalPresentationController: UIPresentationController {
...
+        transition(to: .shortForm)
+    }
 }
 
 // MARK: - Public Methods
 
 public extension PanModalPresentationController {
+    
+    func observeKeyboard() {
+        NotificationCenter.default.addObserver(self, selector: #selector(self.keyboardWillShow), name: UIResponder.keyboardWillShowNotification, object: nil)
```
Comment (markj) [inline PanModal/Controller/PanModalPresentationController.swift:316]: You need to unsubscribe from these.

**Review by markj — CHANGES_REQUESTED** [review]

File: PanModal/Controller/PanModalPresentationController.swift (hunk 3 above)
Comment (markj) [reply PanModal/Controller/PanModalPresentationController.swift:316]: I'm wrong... you don't need to anymore.. I thought that was only the token version.

File: PanModal/Controller/PanModalPresentationController.swift [hunk 4]
```diff
@@ -234,12 +256,64 @@ open class PanModalPresentationController: UIPresentationController {
...
+    @objc func keyboardWillShow(notification: Notification) {
+        
+        //if the presented view is already at max height and is scrollable then scroll to above the keyboard when keyboard is shown if needed
+        
+        if let keyboardSize = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect {
+            keyboardHeight = keyboardSize.height
+            
+            if let scrollView = presentable?.panScrollable, pannedToMax {
+                currentScrollContentInset = scrollView.contentInset.bottom + keyboardHeight
+                scrollView.contentInset = UIEdgeInsets(top: 0.0, left: 0.0, bottom: currentScrollContentInset, right: 0.0);
```
Comment (markj) [inline PanModal/Controller/PanModalPresentationController.swift]: This shouldn't have a semi colon.

you should run swift lint

**Comment by markj** [comment]
I *think* you need to be updating the scrollbar insets as well as the content insets.

**Pull request closed without merging**